components/events/event.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <div class="card shadow-2">
+    <img :src="img" />
+    <h2>{{ title }}</h2>
+    <div class="content">
+      <div class="compact">
+        <div class="detail">
+          <img src="/images/browser/calendar.svg" class="icon" />
+          <div class="text">
+            <h3>When</h3>
+            <p>{{ date }}</p>
+          </div>
+        </div>
+
+        <div class="detail">
+          <img src="/images/browser/user.svg" class="icon" />
+          <div class="text">
+            <h3>Organizer</h3>
+            <p>{{ organizer }}</p>
+          </div>
+        </div>
+      </div>
+
+      <div class="detail">
+        <img src="/images/browser/list.svg" class="icon" />
+        <div class="text">
+          <h3>Activities</h3>
+          <ul>
+            <li v-for="activity in activities.slice(0, 3)" :key="activity">
+              {{ activity }}
+            </li>
+            <li v-if="activities.length > 3" class="extra">+ {{ activities.length - 3 }} more</li>
+          </ul>
+        </div>
+      </div>
+      <a :href="url" class="btn filled green shadow-1">see details</a>
+    </div>
+  </div>
+</template>
+
+<script>
+  import Vue from 'vue';
+  export default Vue.extend({
```

Comment:
just `export default {}`

Reply:
Both are effectively equivalent, but this one helps my IDE (VS Code) with typings (even if it is not Typescript) which I prefer.

Reply:
IDE is changable, but code should look pretty. This type requires one extra useless import

Reply:
Hate to intervene when I don't really know sh!t about what's going on, but typings don't hurt code quality. Excess imports do, however. Any way to preserve the typings by just exporting the object with the `props` property, without throwing it on top of the Vue object?

Reply:
@illright, by typings I didn't mean the props or the component's interface, but rather while developing getting helpful auto-complete options. Also, the imports all share the same object, so no extra load is added to the build.
However, if you don't like it I'll remove it.

Reply:
@aabounegm yeah, that's what I meant. I see that most type information is stored in the object you're extending `Vue` with, so any way to preserve that information but without the Vue object?
For a second thought, if you say that all imports share the same object (which is kind of strange, but could be the case), then I'm ok with the import and the typing

Reply:
Without actually using Typescript (which I voted for in the beginning), there is no other way. VS Code does provide some hints even with `export default {};`, though not as good. But if this is the style we agreed on, then I don't mind.
As for the import thing, all imports do indeed share the same object. This is the case with all Javascript imports (be it `require` in Node, ES6 imports, CommonJS libraries, ...)